onnx/optimizer/passes/optimize_pass.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "onnx/onnx_pb.h"
+#include "onnx/ir.h"
+
+namespace onnx { namespace optimization {
+
+enum class API_TYPE {
+  proto, ir
+};
+
+struct OptimizePass {
+
+  std::string name;
+  API_TYPE type;
+
+  explicit OptimizePass(const std::string name, API_TYPE type)
+    : name(name), type(type) {
+  }
+
+  virtual void optimize(onnx::ModelProto& mp) {}
+
+  virtual void optimize(std::shared_ptr<Graph>& graph) {}
```

Comment:
The point that we keep both APIs (proto and Ir in-memory) is that, the ir in-memory API design may not be good enough for optimization, right?

My opinion is that we should use the in-memory one as long as IR APIs are designed (the goal of IR APIs design is to be good enough for optimization and execution).

Reply:
@bddppq suggests to support ModelProto as input. So if someone wants to directly operate on Protobuf, just let them do.

Reply:
Hmmm, I do hold different opinion :). If someone wants to directly operate on Protobuf, it means our ir API is not good (even worse than protobuf :)), which should not be the case, and even if it's the case, community should keep tuning the API till it's good. :) Anyway, let's not be blocked on this for now.